docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/git-merge-pr.md

```diff
@@ -10,6 +10,20 @@ description: Merges an open pull request.
 `git-merge-pr` merges an open pull request. This step commonly follows a
 [`git-open-pr`](git-open-pr.md) step.
 
+:::caution
+This step only supports synchronous merges due to API limitations of the Git providers.
+It does not work with merge queues or wait for queued merges to finish. If a
+repository enforces merge queues, the behavior depends on the token permissions.
+
+- GitHub: If a merge queue is required, the PR will either be merged directly if the
+  token has bypass permissions, or it will fail with an error if it does not.
+- GitLab: If a merge train is required, the merge will either fail or can be bypassed
+  based on permissions.
```

Comment:
GitLab actually has no feature that requires PRs to be merged using a merge train. It is always optional.

Reply:
Nevermind what the API can or cannot do. GitLab has no way of _requiring_ that MRs be merged using a merge train. So yes. You can remove that warning because its entire premise is incorrect.